lib/puppet/functions/simplib/to_integer.rb

```diff
@@ -0,0 +1,21 @@
+# Converts the argument into an `Integer`.
+# Only works if the passed argument responds to the `to_i()` Ruby method.
+#
+Puppet::Functions.create_function(:'simplib::to_integer') do
```

Comment:
For these conversion functions, I think we should encourage use of [assrt_type](https://docs.puppet.com/puppet/latest/function.html#asserttype) or regular type manipulation listed in each data type's docs page ([string](https://docs.puppet.com/puppet/5.0/lang_data_number.html#converting-strings-to-numbers) for example) and the old functions should just be deprecated and abandoned.